### PR TITLE
patch: did issue stuff, it works

### PR DESCRIPTION
**Pull request by Radiump123**
modified:   pyproject.toml
	modified:   src/rufus_py/drives/find_usb.py
	modified:   src/rufus_py/drives/formatting.py
	modified:   tests/test_formatting.py
	modified:   uv.lock

it works, i did what was asked in the issues, mostly

## Summary by Sourcery

Improve USB drive formatting robustness and add partitioning support based on application state.

New Features:
- Add partition scheme application before formatting, supporting GPT and MBR via a new helper that operates on the raw device.
- Introduce a bad-block checking routine that runs badblocks with configurable passes based on state and reports results.

Bug Fixes:
- Ensure drive node and mount resolution is done lazily via a helper function instead of at import time, preventing crashes when no USB is present.
- Make cluster size computation and label-changing functions handle missing drives and unknown filesystem types gracefully, always returning safe defaults or exiting cleanly.
- Update USB device node discovery to return None when no device is found, avoiding implicit fallthrough behavior.

Enhancements:
- Refine formatting commands to log clearer, filesystem-specific debug messages and handle missing system utilities like mkfs tools and parted more explicitly.
- Centralize mount and drive lookup logic into a reusable helper to simplify unmount, remount, formatting, labeling, and repair operations.
- Improve USB detection to better enumerate user media paths and fall back to directory names when LABEL lookup fails.
- Add tests covering partition scheme selection, cluster size behavior, bad-block checks, and resilience of formatting utilities when devices are absent.

Build:
- Add a dev dependency group with pytest to support running the new and existing tests.

Tests:
- Extend formatting tests to assert correct mkfs tool selection alongside partitioning calls and to verify cluster, partitioning, bad-block checking, and labeling helpers behave safely under edge conditions.

### CODE REVIEW

File: src/rufus_py/drives/formatting.py [hunk 1]
```diff
@@ -3,239 +3,278 @@
...
+        print("Error: No drive node found. Cannot check for bad blocks.")
+        return False
+
+    passes = 2 if states.check_bad else 1
+    # -s = show progress, -v = verbose output
+    # -n = non-destructive read-write test (safe default)
+    args = ["badblocks", "-sv", "-b", "4096"]
+    if passes > 1:
+        args.append("-n")  # non-destructive read-write
+    args.append(drive)
```
[the remark this answers is not in the document]
Comment (R-Larocque) [reply src/rufus_py/drives/formatting.py:200]: tabarnak

File: src/rufus_py/drives/formatting.py [hunk 2]
```diff
@@ -3,239 +3,278 @@
...
-    # REMOUNT
+
+
+def _apply_partition_scheme(drive: str):
+    """Write a GPT or MBR partition table to the raw disk.
+
+    states.partition_scheme: 0 = GPT, 1 = MBR
+    states.target_system:    0 = UEFI (non CSM), 1 = BIOS (or UEFI-CSM)
+    """
+    raw_device = drive.rstrip("0123456789")
```
[the remark this answers is not in the document]
Comment (Splayer0) [reply src/rufus_py/drives/formatting.py]: NVMEs arent meant for flashing isos...
